src/views/Signup.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import { commitMutation } from 'react-relay'
+import environment from '../config/relay'
+import { Redirect } from "react-router";
+import graphql from 'babel-plugin-relay/macro'
+
+import {Form, Button, Container} from "react-bootstrap";
+
+const mutation = graphql`
+    mutation SignUpMutation($input: UserCreateInput!){
+        userCreate(input: $input){
+            user {
+                id
+            }
+        }
+    }
+`
+
+export default function SignUp() {
+    const [username, setUsername] = useState("");
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [redirect, setRedirect] = useState(false)
+
+  function validateForm() {
+    return email.length > 0 && password.length > 0;
+  }
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    
+    commitMutation(
```

Comment:
This should use the `mutate` utility that you've written instead of calling `commitMutation` directly. We're going to be writing mutations all over the place, so it's best to be consistent and be able to write it without importing the `environment` in every call. Ping me if that's unclear.